pkg/cloudprovider/cloudprovider.go

```diff
@@ -105,17 +105,17 @@ func (k *k3s) Initialize(clientBuilder cloudprovider.ControllerClientBuilder, st
 		k.workqueue = workqueue.NewRateLimitingQueue(workqueue.DefaultControllerRateLimiter())
 
 		if err := k.Register(ctx, coreFactory.Core().V1().Node(), lbCoreFactory.Core().V1().Pod(), lbDiscFactory.Discovery().V1().EndpointSlice()); err != nil {
-			logrus.Fatalf("Failed to register %s handlers: %v", controllerName, err)
+			logrus.Panicf("failed to register %s handlers: %v", controllerName, err)
```

Comment:
Why do we need to panic rather than fatal?

Reply:
If we panic here, we can `recover` in the `CloudControllerManager` and provide extra context for which component failed. This is more consistent with how the native Kubernetes components handle errors during initialization.

Reply:
Solid.